src/Core/Framework/Plugin/PluginLifecycleService.php

```diff
@@ -144,7 +144,7 @@ public function installPlugin(PluginEntity $plugin, Context $shopwareContext): I
         // TODO NEXT-1797: Not usable with Composer 1.8, Wait for Release of Composer 2.0
         if (next1797()) {
             $this->executor->require($plugin->getComposerName());
-        } else {
+        } elseif ($plugin->isManagedByComposer()) {
```

Comment:
@mitelg the validation is always required, isn't it? Even if it's coming from custom/plugins.

Reply:
yes it is. we tried to validate all plugins with the "require" definition from the composer.json file. But @fixpunkt and I figured out some issues on the dev-days and the whole validation needs some optimisation :wink:

Reply:
When installing a plugin from `custom/plugins`, validating that plugin means that it cannot require any extra libraries and ship those in its own `vendor/` folder.

With this validation in place for plugins from `custom/plugins`, for example, we cannot ship e.g. any shipping integrations as ZIPs over the store, since they need to bundle  https://github.com/VIISON/AddressSplitting and NEXT-1797 likely won't be active for the initial release.

So I'd disagree, validating requirements like this is only an option if we can be certain the plugin was actually `composer require`'d, which is not the case for plugins which were installed into `custom/plugins`, e.g. from the store.

Reply:
but do we need an additional validation if the plugin was installed via composer? on "require" composer will validate and resolve all requirements.

besides that, the validator still needs to consider if a required plugin is installed and active

Reply:
> but do we need an additional validation if the plugin was installed via composer? on "require" composer will validate and resolve all requirements.

No, other than the validation you already mentioned. But that's not the issue at hand: Validating plugins which are **not** required using composer fails their installation if there's anything other than `shopware/platform` in their `require` section. 

So yeah, you're right: The call to `$this->requirementValidator->validateRequirements();` should probably _never_ happen until  Validating requirements should probably _never_ happen until `NEXT-1797` is active, since:

- For non-composer-managed plugins in `custom/plugins`, it will prevent legitimate use cases (shipping extra libraries), and
- For composer-managed plugins, it won't do anything that is not already done by composer.

So the correct solution would be to remove the `else` branch entirely, wouldn't it?

Reply:
> So the correct solution would be to remove the `else` branch entirely, wouldn't it?

no, because then, we would have no validation at all :smiley: 
I just started a ticket on plugin requirements check, so I will consider your cases and ours, so that all plugins should be checked correct :blush: